docs/user_guide/expand_path.rst

```diff
@@ -71,6 +71,141 @@ If your session start time is present in your data path, you can indicate this f
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The IBL's Brain Wide Map features data from several labs of mice performing a visual decision-making task. Some
+experimental sessions, such as those from the Steinmetz Lab, include video recordings of the experiments from three
+cameras, stored in the following directory structure.
+
+.. code-block:: bash
+
+    steinmetzlab/
+    ¦   Subjects/
+    ¦   +-- NR_0017/
+    ¦   ¦   +-- 2022-03-22/
+    ¦   ¦   ¦   +-- 001/
+    ¦   ¦   ¦   ¦   +-- raw_video_data/
+    ¦   ¦   ¦   ¦   ¦   +-- _iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4
+    ¦   ¦   ¦   ¦   ¦   +-- ...
+    ¦   ¦   ¦   ¦   +-- ...
+    ¦   +-- <subject_id>/
+    ¦   ¦   +-- <session_start_time>/
+    ¦   ¦   ¦   +-- 001/
+    ¦   ¦   ¦   ¦   +-- raw_video_data/
+    ¦   ¦   ¦   ¦   ¦   +-- _iblrig_leftCamera.raw.<session_id>.mp4
+    ¦   ¦   ¦   ¦   ¦   +-- ...
+    ¦   ¦   ¦   ¦   +-- ...
+    ¦   ...
+
+We can use :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander` to find these left camera video files and
+extract the subject ID, the session start time (formatted as ``yyyy-mm-dd``), and a hash that we'll treat as
+the session ID.
+
+.. code-block:: python
+
+    >>> source_data_spec = {
+    ...     "IBL": {
+    ...         "base_directory": "/steinmetzlab/",
+    ...         "file_path": "Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/001/raw_video_data/_iblrig_leftCamera.raw.{session_id}.mp4"
+    ...     }
+    ... }
+    >>> metadata_list = path_expander.expand_paths(source_data_spec)
+    >>> for metadata in metadata_list:
+    ...     pprint(metadata)
+    {'metadata': {'NWBFile': {'session_id': '6252a2f0-c10f-4e49-b085-75749ba29c35',
+                             'session_start_time': datetime.datetime(2022, 3, 22, 0, 0)},
+                             'Subject': {'subject_id': 'NR_0017'}},
+     'source_data': {'IBL': {'file_path': 'steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4'}}}
+    ...
+
+If you'd like to experiment with :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander` locally, you can use
+the following commands on Linux to generate some dummy files mimicking the structure of the IBL data shown above.
+
+.. code-block:: bash
+
+    mkdir -p steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/
+    mkdir -p steinmetzlab/Subjects/NR_0019/2022-04-29/001/raw_video_data/
+    mkdir -p steinmetzlab/Subjects/NR_0019/2022-05-03/001/raw_video_data/
+    touch steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4
+    touch steinmetzlab/Subjects/NR_0019/2022-04-29/001/raw_video_data/_iblrig_leftCamera.raw.9041b63e-02e2-480e-aaa7-4f6b776a647f.mp4
+    touch steinmetzlab/Subjects/NR_0019/2022-05-03/001/raw_video_data/_iblrig_leftCamera.raw.05576d90-6fb7-4aba-99ae-7ba63cc50a9a.mp4
```

Comment:
I was thinking more along the lines of a one-line NeuroConv helper that can be called from any platform

Such as

```python
from neuroconv.utils import generate_path_expander_demo_ibl

generate_path_expander_demo_ibl(folder_path="path/to/generate/dummy/files")
```

Reply:
Largely unrelated to this specific suggestion, but looking closer at the IBL data, the hashes in the video file names aren't the same even within the same session (see [here](https://ibl.flatironinstitute.org/public/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/), for example), so I assume they aren't actually session IDs? 

In this case, would it be proper `LocalPathExpander` usage to use empty braces instead? i.e. `/steinmetzlab/Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/001/raw_video_data/_iblrig_leftCamera.raw.{}.mp4`

Reply:
> Largely unrelated to this specific suggestion, but looking closer at the IBL data, the hashes in the video file names aren't the same even within the same session (see [here](https://ibl.flatironinstitute.org/public/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/), for example), so I assume they aren't actually session IDs?

Haha, wow good catch lol

You are absolutely 100% correct; technically speaking IBL has two caveats.

(a) each object has a separate ID, those IDs are then mapped to a session ID (called an `eid` over there) and in the context of code that would go after the `PathExpander` demo of this section you would use a reverse mapping from these object IDs to go back to the `eid`

(b) technically the preferred access method to the content is via `ONE` which only bothers with `eid` (and `pid` for ecephys data) and handles all this under the hood

But here's the thing, we're kind of strapped for nice but complicated public examples of this kind of thing because most of our actual cases are only for privately shared data so for all intents and purposes I'd say it's fine to demo it's usage portraying them as IDs; or if the blank braces actually works in the `PathExpander` code, that's fine too